src/BugReporting.jl

```diff
@@ -393,11 +384,7 @@ function replay(trace_url=default_rr_trace_dir(); gdb_commands=[], gdb_flags=``,
     end
 
     # replay with rr
-    proc = rr() do rr_path
-        gdb() do gdb_path
-            run(`$rr_path replay $rr_replay_flags -d $gdb_path $trace_dir -- $gdb_args`)
-        end
-    end
+    proc = run(`$(rr()) replay $rr_replay_flags -d $(gdb().exec) $trace_dir -- $gdb_args`)
```

Comment:
Is it ok to just ignore the env for `gdb` here or do we need to manually merge the envs for `rr` and `gdb`?

Reply:
Sadly, we do need to merge the envs, as both `rr` and `gdb` need `LD_LIBRARY_PATH` properly set..  Alternatively, you can fix https://github.com/JuliaLang/julia/issues/39282 and this will "just work".

Reply:
Hmm, I thought `addenv(rr(), "PATH" => GDB_jll.PATH[])` could be used to merge paths but apparently it just overrides `PATH`

Reply:
If you need inspiration: https://github.com/JuliaPackaging/BinaryBuilderBase.jl/blob/b52b7a18bddda8c9a42fbd819b06a4183386bf5a/src/Prefix.jl#L305-L327

Reply:
Ok, I have no idea what `addenv` acually does here:
```julia
julia> path(x) = x.env[findall(startswith("PATH"), x.env)[]]
path (generic function with 1 method)

julia> path(addenv(rr(), "PATH" => GDB_jll.PATH[]))
"PATH=/home/simeon/.julia/artifacts/2bce74229f51de64c33433272240af5734619b33/bin:/home/simeon/.julia/artifacts/d6e550073d22b9b1976e67019e04eaa06f5f69f6/bin:/home/simeon/.julia/artifacts/69903d1a57355b6e78750aac3a8b3fddab87180f/bin:/home/simeon/.julia/artifacts/51cb236ffdb7e1ed1b9d44f14c81f2b84bc46520/bin:/home/simeon/.julia/artifacts/6d2f6b6f4f4cb56741ee630fbec4d86be20cbab7/bin"

julia> path(rr())
"PATH=/home/simeon/.julia/artifacts/5bac04ea38e25b2864bbbd023944b7467f0ba694/bin:/home/simeon/.local/bin:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl"

julia> path(gdb())
"PATH=/home/simeon/.julia/artifacts/2bce74229f51de64c33433272240af5734619b33/bin:/home/simeon/.julia/artifacts/d6e550073d22b9b1976e67019e04eaa06f5f69f6/bin:/home/simeon/.julia/artifacts/69903d1a57355b6e78750aac3a8b3fddab87180f/bin:/home/simeon/.julia/artifacts/51cb236ffdb7e1ed1b9d44f14c81f2b84bc46520/bin:/home/simeon/.julia/artifacts/6d2f6b6f4f4cb56741ee630fbec4d86be20cbab7/bin:/home/simeon/.local/bin:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl"
```